src/SimSpace/SimFormat.hs

```diff
@@ -370,13 +383,85 @@ reformat programLines =
     processBlock (Left  s) = pure (Left s)
     processBlock (Right s) = Right <$> parseMaybe parseImportBlock s
 
-    reassemble :: [String]
-               -> ([Either BlankLine SortedImportStmts], [Block])
-               -> [Block]
-    reassemble nonImports (chunkedImports, leftovers)
-      = fmap Left nonImports
-     <> (fmap.fmap) renderImportStmts chunkedImports
-     <> leftovers
+    reassemble
+      :: (MonadIO m)
+      => [String]
+      -> ([Either BlankLine SortedImportStmts], [Block])
+      -> m [Block]
+    reassemble nonImports (chunkedImports, leftovers) = do
+        rechucked <- (if regroup then rechunk else pure) chunkedImports
```

Comment:
you probably meant "rechu*n*cked", not "rechucked"

Reply:
fixed in 00c1135